main.tf

```diff
@@ -0,0 +1,59 @@
+module "label" {
+  source     = "git::https://github.com/cloudposse/terraform-null-label.git?ref=tags/0.3.3"
+  namespace  = "${var.namespace}"
+  stage      = "${var.stage}"
+  name       = "${var.name}"
+  delimiter  = "${var.delimiter}"
+  attributes = "${var.attributes}"
+  tags       = "${var.tags}"
+}
+
+module "kops_metadata" {
+  source       = "git::https://github.com/cloudposse/terraform-aws-kops-metadata.git?ref=tags/0.1.1"
+  dns_zone     = "${var.cluster_name}"
+  masters_name = "${var.masters_name}"
+  nodes_name   = "${var.nodes_name}"
+}
+
+resource "aws_iam_role" "default" {
+  name        = "${module.label.id}"
+  description = "Role that can be assumed by atlantis"
+
+  lifecycle {
+    create_before_destroy = true
+  }
+
+  assume_role_policy = "${data.aws_iam_policy_document.assume_role.json}"
+}
+
+data "aws_iam_policy_document" "assume_role" {
+  statement {
+    actions = [
+      "sts:AssumeRole",
+    ]
+
+    principals {
+      type        = "Service"
+      identifiers = ["ec2.amazonaws.com"]
+    }
+
+    principals {
+      type = "AWS"
+
+      identifiers = [
+        "${module.kops_metadata.masters_role_arn}",
```

Comment:
why masters? or only masters?
do we want just nodes to assume the role?

Reply:
with `kiam` you *only* want to grant access to masters.